pkg/controllers/dashboard/helm/repo.go

```diff
@@ -227,26 +233,39 @@ func (r *repoHandler) download(repoSpec *catalog.RepoSpec, status catalog.RepoSt
 	}
 
 	downloadTime := metav1.Now()
-	if repoSpec.GitRepo != "" && status.IndexConfigMapName == "" {
-		commit, err = git.Head(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify, repoSpec.CABundle)
-		if err != nil {
-			return status, err
-		}
-		status.URL = repoSpec.GitRepo
-		status.Branch = repoSpec.GitBranch
-		index, err = git.BuildOrGetIndex(metadata.Namespace, metadata.Name, repoSpec.GitRepo)
-	} else if repoSpec.GitRepo != "" {
-		commit, err = git.Update(secret, metadata.Namespace, metadata.Name, repoSpec.GitRepo, repoSpec.GitBranch, repoSpec.InsecureSkipTLSverify, repoSpec.CABundle)
+	if repoSpec.GitRepo != "" {
```

Comment:
This changes the behavior I think and we should stick to old behavior as this PR should be only be a replacement of git package. Can we please change back the if conditions as before and can you make a new PR with the behavioural changes ?

Reply:
This is the same behavior.  I am 100% sure.

Reply:
I am adding comments to make it easier to understand.

Reply:
Gotcha. You are right